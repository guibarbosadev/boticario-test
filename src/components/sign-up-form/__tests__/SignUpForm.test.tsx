import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import SignUpForm, { SignUpFormProps } from '..';

describe('SignUpForm | Component', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<SignUpForm />);

        expect(toJSON()).toMatchSnapshot();
    });

    it('renders name, email and password inputs', () => {
        const { queryByPlaceholder } = render(<SignUpForm />);
        const placeholders = ['Nome', 'Email', 'Senha'];

        placeholders.forEach((placeholder) => {
            expect(queryByPlaceholder(placeholder)).toBeTruthy();
        });
    });

    it('renders submit button', () => {
        const { queryByTestId } = render(<SignUpForm />);

        expect(queryByTestId('submit-button')).toBeTruthy();
    });

    it('renders initial values', () => {
        const props: SignUpFormProps = {
            initialValues: {
                name: 'Some name',
                email: 'email@example.com',
                password: 'some password'
            }
        };
        const { queryByDisplayValue } = render(<SignUpForm {...props} />);

        for (const key in props.initialValues) {
            const value = props.initialValues[key];
            expect(queryByDisplayValue(value)).toBeTruthy();
        }
    });

    it('[VALID] call props.onSubmit', () => {
        const props: SignUpFormProps = {
            initialValues: {
                name: 'Some name',
                email: 'email@example.com',
                password: 'some password'
            },
            onSubmit: jest.fn()
        };
        const { getByTestId } = render(<SignUpForm {...props} />);
        const submitButton = getByTestId('submit-button');

        expect(props.onSubmit).toHaveBeenCalledTimes(0);

        fireEvent.press(submitButton);
        expect(props.onSubmit).toHaveBeenCalledTimes(1);
    });

    it('[INVALID] DO NOT calls props.onSubmit', () => {
        const props: SignUpFormProps = {
            onSubmit: jest.fn()
        };
        const { getByTestId } = render(<SignUpForm {...props} />);
        const submitButton = getByTestId('submit-button');

        expect(props.onSubmit).toHaveBeenCalledTimes(0);

        fireEvent.press(submitButton);
        expect(props.onSubmit).toHaveBeenCalledTimes(0);
    });
});
