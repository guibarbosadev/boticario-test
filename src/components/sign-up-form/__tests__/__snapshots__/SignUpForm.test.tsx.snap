// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm | Component renders correctly 1`] = `
Array [
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Nome"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />,
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    textContentType="emailAddress"
    underlineColorAndroid="transparent"
    value=""
  />,
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Senha"
    rejectResponderTermination={true}
    textContentType="password"
    underlineColorAndroid="transparent"
    value=""
  />,
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submit-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Cadastrar-se
      </Text>
    </View>
  </View>,
]
`;
